src/data/metadata.ts

```diff
@@ -54,32 +62,39 @@ function guid() {
   let uuid = "xxxxxxxxxxxx4xxxyxxxxxxxxxxxxxxx".replace(/[xy]/g, function(c) {
     let r = (d + Math.random() * 16) % 16 | 0;
     d = Math.floor(d / 16);
-    return (c == "x" ? r : (r & 0x3 | 0x8)).toString(16);
+    return str((c == "x" ? r : (r & 0x3 | 0x8)), 16);
   });
   return uuid;
 }
 // tslint:enable
 
-function track(data: CookieInfo): void {
-  let expiry = new Date();
-  expiry.setDate(expiry.getDate() + config.expire);
-  let expires = expiry ? "expires=" + expiry.toUTCString() : "";
-  let value = `${data.userId}|${data.sessionId}|${data.timestamp}` + ";" + expires + ";path=/";
-  document.cookie = CLARITY_COOKIE_NAME + "=" + value;
+function session(ts: number): string {
+  let id = str(ts, 36);
+  if (config.track && sessionStorage) {
+    let value = sessionStorage.getItem(CLARITY_STORAGE_KEY);
+    if (value && value.indexOf(CLARITY_STORAGE_SEPARATOR) >= 0) {
+      let parts = value.split(CLARITY_STORAGE_SEPARATOR);
+      if (parts.length === 2 && ts - parseInt(parts[1], 10) < config.session) { id = parts[0]; }
+    }
+    sessionStorage.setItem(CLARITY_STORAGE_KEY, `${id}|${ts}`);
+  }
+  return id;
+}
+
+function str(number: number, base: number = 10): string {
+  return number.toString(base);
 }
 
-function read(): CookieInfo {
+function user(): string {
+  let id = guid();
   let cookies: string[] = document.cookie.split(";");
   if (cookies) {
     for (let i = 0; i < cookies.length; i++) {
       let pair: string[] = cookies[i].split("=");
-      if (pair.length > 1 && pair[0].indexOf(CLARITY_COOKIE_NAME) >= 0 && pair[1].indexOf(CLARITY_COOKIE_SEPARATOR) > 0) {
-        let parts = pair[1].split(CLARITY_COOKIE_SEPARATOR);
-        if (parts.length === 3) {
-          return { userId: parts[0], sessionId: parts[1], timestamp: parseInt(parts[2], 10) };
-        }
+      if (pair.length > 1 && pair[0].indexOf(CLARITY_STORAGE_KEY) >= 0 && pair[1].length === 32) {
```

Comment:
>CLARITY_STORAGE_KEY [](start = 45, length = 19)

Shouldn't it generate the guid() only when it fails to retrieve it? Optimization only -- looks correct as written.

Reply:
Fixed, thanks.